snappi/snappicommon.py

```diff
@@ -141,15 +237,34 @@ def parent(self):
         return self._parent
         
     def _get_property(self, name, default_value=None, parent=None, choice=None):
-        if name not in self._properties or self._properties[name] is None:
-            if isinstance(default_value, type) is True:
-                self._properties[name] = default_value(parent=parent, choice=choice)
+        if name in self._properties and self._properties[name] is not None:
+            return self._properties[name]
+        if isinstance(default_value, type) is True:
+            if 'choice' in dir(self):
+                self._properties['choice'] = name
+            self._properties[name] = default_value(parent=parent, choice=choice)
+
+            if '_DEFAULTS' in dir(self._properties[name]) and\
+                'choice' in self._properties[name]._DEFAULTS:
+                # this node is used to set the if the choice has default.
```

Comment:
this is not a good side effect and can be mitigated. i.e. whenever a choice is set to certain enum value, the attributes pertaining to all other enum values can be set to `None`

Reply:
will raise a issue for this, need more analysis to fix this issue

Reply:
coded changes done in latest revision